docs/source/examples/pt_zne.md

```diff
@@ -0,0 +1,400 @@
+            )
+
+    return ptm_matrix
+```
+
+Let us consider a simple circuit of a CNOT gate. We are going to subject this circuit to two types of noise and compare
+their respective PTMs.
+
+```{code-cell} ipython3
+import numpy as np
+import seaborn as sns
+import matplotlib.pylab as plt
+
+from cirq import LineQubit, Circuit, CNOT, Ry, depolarize, X, Y, Z
+
+q0 = LineQubit(0)
+q1 = LineQubit(1)
+
+circuit = Circuit(CNOT(q0, q1))
+print(circuit)
+```
+```{code-cell} ipython3
+ptmcnot = ptm_matrix(circuit, 2)
+ax = sns.heatmap(ptmcnot.real, linewidth=0.5, vmin=-1, vmax=1, cmap="PiYG")
+ax.set_title('Ideal CNOT PTM')
+plt.show()
+```
+```{code-cell} ipython3
+# PTM of a noisy CNOT gate: depolarizing noise
+noisy_circuit_incoherent = circuit.with_noise(depolarize(p=0.3))
+print(noisy_circuit_incoherent)
+
+ptmcnot = ptm_matrix(noisy_circuit_incoherent, 2)
+ax = sns.heatmap(ptmcnot.real, linewidth=0.5, vmin=-1, vmax=1, cmap="PiYG")
+ax.set_title('PTM of noisy CNOT (incoherent)')
+plt.show()
+```
+```{code-cell} ipython3
+# PTM of a noisy CNOT gate: Rz
+noisy_circuit_coherent = circuit.with_noise(Ry(rads=np.pi/12))
+print(noisy_circuit_coherent)
+
+ptmcnot = ptm_matrix(noisy_circuit_coherent, 2)
+ax = sns.heatmap(ptmcnot.real, linewidth=0.5, vmin=-1, vmax=1, cmap="PiYG")
+ax.set_title('PTM of noisy CNOT (coherent)')
+plt.show()
+```
+If we compare the PTM of the ideal CNOT gate to those when the gate was subjected to incoherent noise and coherent noise, 
+there are additional sources of errors to deal with when coherent noise is acting on the CNOT gate. These can be reduced or tailored to be close to how the incoherent noise PTM appears through Pauli Twirling.
+
+```{code-cell} ipython3
+from mitiq.pt import generate_pauli_twirl_variants
+
+# Generate twirled circuits
+NUM_TWIRLED_VARIANTS = 3
+twirled_circuits = generate_pauli_twirl_variants(
+    circuit, num_circuits=NUM_TWIRLED_VARIANTS)
+print("Example ideal twirled circuit", twirled_circuits[-1], sep="\n")
+```
+Now, lets add coherent noise to the CNOT gate in each twirled circuit.
+```{code-cell} ipython3
+
+noisy_twirled_circuits = []
+
+for circ in twirled_circuits:
+    split_circuit = Circuit(circ[0], circ[1], Ry(rads=np.pi/12)(q0), Ry(rads=np.pi/12)(q1), circ[-1])
+    noisy_twirled_circuits.append(split_circuit)
+
+print("Example noisy twirled circuit", noisy_twirled_circuits[-1], sep="\n")
+```
+
+The twirled PTM is averaged over each noisy twirled circuit such that the new PTM is close to that of the PTM of incoherent noise. We skip the step in this section as we require a very large number of twirled circuits to demonstrate the desired effect of averaging over multiple numpy arrays. The variations in pauli twirled PTMs are shown below when averaged over a different number of pauli twirled circuits: [3](../img/pt_zne_3_circuits.png), [5](../img/pt_zne_5_circuits.png), [30](../img/pt_zne_30_circuits.png) and [300](../img/pt_zne_100_circuits.png).  
```

Comment:
IDK if there's a better way to provide a link to a couple of png images. These appear as images a user will have to download. 

![image](https://github.com/user-attachments/assets/b35b2fb1-dc37-4009-856a-a8941b9744e9)